src/OrchardCore.Modules/OrchardCore.Forms/Migrations.cs

```diff
@@ -149,6 +141,28 @@ public int UpdateFrom2()
 
             return 3;
         }
+        public int UpdateFrom3()
+        {
+            //ValidationRule
+            _contentDefinitionManager.AlterPartDefinition("ValidationRulePart", part => part
+                .WithDescription("Provides the ability to use and configure validation rules for form fields.")
+            );
+
+
+            _contentDefinitionManager.AlterTypeDefinition("Input", type => type
+                .WithPart("ValidationRulePart", part => part
+                    .WithPosition("3")
+                 )
+             );
+
+            _contentDefinitionManager.AlterTypeDefinition("TextArea", type => type
+                .WithPart("ValidationRulePart", part => part
+                    .WithPosition("3")
+                )
+            );
+
+            return 4;
+        }
```

Comment:
This is good like this but you also need to add the same migration logic back to `Create()` too so fresh installs don't have to execute the separate update step. Here it is explained why this is useful: https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/dev/Migrations/BookMigrations.cs#L46

Reply:
`ValidationRulePart`'s description being added is missing from `Create()`.